Tests/TracingInstrumentationTests/TracedLockTests.swift

```diff
@@ -90,15 +90,15 @@ private final class TracedLockPrintlnTracer: TracingInstrument {
 
     struct TracedLockPrintlnSpan: Span {
         let operationName: String
-        let kind: SpanKind
+        private let kind: SpanKind
 
-        var status: SpanStatus? {
+        private var status: SpanStatus? {
             didSet {
```

Comment:
Nitpick: Move this into `setStatus`?

Reply:
Changed as requested, although either I misunderstood it or its opposite to OTel spec:

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#isrecording

> This flag SHOULD be used to avoid expensive computations of a Span attributes or events in case when a Span is definitely not recorded. Note that any child span's recording is determined independently from the value of this flag (typically based on the sampled flag of a TraceFlag on SpanContext).

> This flag may be true despite the entire trace being sampled out. This allows to record and process information about the individual Span without sending it to the backend. An example of this scenario may be recording and processing of all incoming requests for the processing and building of SLA/SLO latency charts while sending only a subset - sampled spans - to the backend. See also the sampling section of SDK design.

> Users of the API should only access the IsRecording property when instrumenting code and never access SampledFlag unless used in context propagators.

My understanding of the text above is that `isRecording` is a flag indicating is the span/segment is sampled;
if it is not sampled then, as with NoOpSpan, there is no point to o any work

the implementation in `TracedLockPrintlnSpan` is using `isRecording` flag to indicate if anything has been recorded.

```swift        
      private var events = [SpanEvent]() {
            didSet {
                self.isRecording = !self.events.isEmpty
            }
        }

        var attributes: SpanAttributes = [:] {
            didSet {
                self.isRecording = !self.attributes.isEmpty
            }
        }
```

Reply:
Oh the traced lock is probably not very correctly implemented, please ignore or fix it. Sorry about that!